File: src/main/java/com/github/albertosh/flitetrakr/util/language/LanguageUtils.java
package com.github.albertosh.flitetrakr.util.language;

import java.util.Locale;
import java.util.ResourceBundle;

public class LanguageUtils {

    private final static String DEFAULT_LANGUAGE = "en";
    private final static String DEFAULT_COUNTRY = "US";

    private static String language = DEFAULT_LANGUAGE;
    private static String country = DEFAULT_COUNTRY;

    private static Locale currentLocale = new Locale(language, country);

    public static void setLanguage(String language, String country) {
        LanguageUtils.language = language;
        LanguageUtils.country = country;
        currentLocale = new Locale(language, country);
    }

    public static String getMessage(Message message) {
        ResourceBundle messages = ResourceBundle.getBundle("messages", currentLocale);
        return messages.getString(message.getKey());
    }
}

File: core_kotlin/src/main/kotlin/com/github/albertosh/flitetrakr/util/language/LanguageUtils.kt
package com.github.albertosh.flitetrakr.util.language

import java.util.*

object LanguageUtils {

    private val DEFAULT_LANGUAGE = "en"
    private val DEFAULT_COUNTRY = "US"

    private var language = DEFAULT_LANGUAGE
    private var country = DEFAULT_COUNTRY

    private var currentLocale = Locale(language, country)

    fun setLanguage(language: String, country: String) {
        LanguageUtils.language = language
        LanguageUtils.country = country
        currentLocale = Locale(language, country)
    }

    fun getMessage(message: Message): String {
        val messages = ResourceBundle.getBundle("messages", currentLocale)
        val key = message.getKey()
        return messages.getString(key)
    }
}
